### PR TITLE
many refactor and style modify

### PR DESCRIPTION
**Pull request by MisileLab**

### CODE REVIEW

File: dislash/interactions/message_components.py [hunk 1]
```diff
@@ -97,20 +96,20 @@ class ButtonStyle:
...
+    success = 3
+    green = 3
 
-    danger    = 4
-    red       = 4
+    danger = 4
+    red = 4
 
-    link      = 5
+    link = 5
```
Comment (m1raynee) [inline dislash/interactions/message_components.py:112]:
```suggestion
    # fmt: off
    primary   = 1
    blurple   = 1

    secondary = 2
    grey      = 2
    gray      = 2

    success   = 3
    green     = 3

    danger    = 4
    red       = 4

    link      = 5
    # fmt: on
```

File: dislash/slash_commands/slash_command.py [hunk 2]
```diff
@@ -27,15 +26,15 @@ class Type:
...
-    CHANNEL           = 7
-    ROLE              = 8
-    MENTIONABLE       = 9
+    STRING = 3
+    INTEGER = 4
+    BOOLEAN = 5
+    USER = 6
+    CHANNEL = 7
+    ROLE = 8
+    MENTIONABLE = 9
```
Comment (m1raynee) [inline dislash/slash_commands/slash_command.py]:
```suggestion
    # fmt: off
    SUB_COMMAND       = 1
    SUB_COMMAND_GROUP = 2
    STRING            = 3
    INTEGER           = 4
    BOOLEAN           = 5
    USER              = 6
    CHANNEL           = 7
    ROLE              = 8
    MENTIONABLE       = 9
    # fmt: on
```

File: dislash/slash_commands/slash_core.py [hunk 3]
```diff
@@ -38,9 +37,9 @@
 )
 
 
-#-----------------------------------+
-#              Utils                |
-#-----------------------------------+
+# -----------------------------------+
+# |             Utils                |
+# -----------------------------------+
```
Comment (m1raynee) [inline dislash/slash_commands/slash_core.py]:
```suggestion
# +-----------------------------------+
# |              Utils                |
# +-----------------------------------+

```

File: dislash/slash_commands/slash_core.py [hunk 4]
```diff
@@ -54,9 +53,9 @@ def get_class(func):
             return getattr(mod, class_name(func), None)
 
 
-#-----------------------------------+
-#         Core and checks           |
-#-----------------------------------+
+# -----------------------------------+
+# |        Core and checks           |
+# -----------------------------------+
```
Comment (m1raynee) [inline dislash/slash_commands/slash_core.py]:
```suggestion
# +-----------------------------------+
# |         Core and checks           |
# +-----------------------------------+

```

File: dislash/slash_commands/slash_core.py [hunk 5]
```diff
@@ -76,12 +75,12 @@ def __init__(self, func, *, name=None, connectors=None, **kwargs):
         if cooldown is None:
             try:
                 # Assuming that it's discord.py 1.7.0+
-                self._buckets = CooldownMapping(cooldown, BucketType.default)
-            except:
+                self._buckets = CooldownMapping(cooldown)
```
Comment (m1raynee) [inline dislash/slash_commands/slash_core.py]:
```suggestion
                self._buckets = CooldownMapping(cooldown, BucketType.default)
```
What is the reason for dropping this?

File: dislash/slash_commands/slash_core.py [hunk 6]
```diff
@@ -466,12 +477,14 @@ def predicate(ctx):
...
     """Similar to ``commands.has_any_role``"""
+
     def predicate(ctx):
         if not isinstance(ctx.channel, discord.abc.GuildChannel):
             raise NoPrivateMessage()
 
         getter = functools.partial(discord.utils.get, ctx.author.roles)
-        if any(getter(id=item) is not None if isinstance(item, int) else getter(name=item) is not None for item in items):
+        if any(getter(id=item) is not None if isinstance(item, int) else getter(name=item) is not None for item in
+               items):
```
Comment (m1raynee) [inline dislash/slash_commands/slash_core.py:487]:
```suggestion
        if any(getter(id=item) is not None if isinstance(item, int) else getter(name=item) is not None
        for item in items):
```
It looks better

File: dislash/slash_commands/slash_core.py [hunk 7]
```diff
@@ -494,21 +507,25 @@ def predicate(ctx):
...
     def predicate(ctx):
         ch = ctx.channel
         if not isinstance(ch, discord.abc.GuildChannel):
             raise NoPrivateMessage()
 
         me = ch.guild.me
         getter = functools.partial(discord.utils.get, me.roles)
-        if any(getter(id=item) is not None if isinstance(item, int) else getter(name=item) is not None for item in items):
+        if any(getter(id=item) is not None if isinstance(item, int) else getter(name=item) is not None for item in
+               items):
```
Comment (m1raynee) [inline dislash/slash_commands/slash_core.py]:
```suggestion
        if any(getter(id=item) is not None if isinstance(item, int) else getter(name=item) is not None
        for item in items):
```
Simmilar

File: dislash/slash_commands/utils.py [hunk 8]
```diff
@@ -121,73 +120,83 @@ def matching_condition(self, check, *, cancel_others: bool=False, reset_timeout:
...
+    def no_checks(self, *, cancel_others: bool = False, reset_timeout: bool = True):
         """
         A decorator that makes the function below waiting for any click.
 
         Parameters are the same as in :meth:`matching_condition`, except
         there's no ``check``.
         """
-        def always(inter):
+
+        def always():
```
Comment (m1raynee) [inline dislash/slash_commands/utils.py:181]: Strange, I think there is a magical meaning in the presence of this argument

File: dislash/slash_commands/slash_client.py [hunk 9]
```diff
@@ -78,47 +78,62 @@ def _register_listeners(self):
...
-                check = lambda inter: True
+                def check(inter):
+                    if message.id != inter.message.id:
+                        return False
+                    else:
+                        return True
+
             def auto_check(inter):
                 if message.id != inter.message.id:
                     return False
```
Comment (m1raynee) [inline dislash/slash_commands/slash_client.py]:
```suggestion
```
Simmalar to [L99-L100](https://github.com/EQUENOS/dislash.py/pull/35/files#diff-3b70d3f8c8ad87ae86db4601fb6edd0980ed2243d9e1da6983ebc6c6f4146055L99-L100)

File: dislash/slash_commands/slash_client.py [hunk 10]
```diff
@@ -1025,7 +1045,11 @@ async def _cache_guild_commands(self):
     # Special waiter
     async def wait_for_button_click(self, check=None, timeout=None):
         if check is None:
-            check = lambda ctx: True
+            def check(inter):
+                if message.id != inter.message.id:
+                    return False
+                else:
+                    return True
```
Comment (m1raynee) [inline dislash/slash_commands/slash_client.py:1050]:
```suggestion
check = lambda ctx: True
```
This is absolutely wrong. `message` out of nowhere?? I think some of the code just migrated here due to the incorrect operation of the automatic parser of your IDE.

File: dislash/slash_commands/slash_client.py [hunk 11]
```diff
@@ -1035,7 +1059,11 @@ async def wait_for_button_click(self, check=None, timeout=None):
 
     async def wait_for_dropdown(self, check=None, timeout=None):
         if check is None:
-            check = lambda ctx: True
+            def check(inter):
+                if message.id != inter.message.id:
+                    return False
+                else:
+                    return True
```
Comment (m1raynee) [inline dislash/slash_commands/slash_client.py:1064]:
```suggestion
check = lambda ctx: True
```
Simmilar to [previous one](https://github.com/EQUENOS/dislash.py/pull/35/files#r671050405)

**Review by m1raynee — CHANGES_REQUESTED** [review]
In general, I noted all the wrong things that I found

**Comment by MisileLab** [comment]
@m1raynee Can you pull request to my fork repository?

File: dislash/slash_commands/slash_client.py (hunk 11 above)
Comment (MisileLab) [reply dislash/slash_commands/slash_client.py:1064]: but pep-8 say do not assign a lambda expression, use a def

File: dislash/slash_commands/slash_client.py (hunk 10 above)
Comment (MisileLab) [reply dislash/slash_commands/slash_client.py:1050]: but pep-8 say do not assign a lambda expression, use a def

Comment (m1raynee) [reply dislash/slash_commands/slash_client.py:1050]: when 
```suggestion
            def check(inter):
                return True
```
The current check function was taken from [R115-R120](https://github.com/EQUENOS/dislash.py/pull/35/files/23d8ca0ae44f06b60e532ed074e4a2f9be3d3b11#diff-3b70d3f8c8ad87ae86db4601fb6edd0980ed2243d9e1da6983ebc6c6f4146055R115-R120) does not fit here.

Comment (m1raynee) [reply dislash/slash_commands/slash_client.py:1050]: I'm tired of explaining to you for the third time what is wrong here. I ask you to carefully reread my comments on GitHub and messages in Discord.

File: dislash/slash_commands/slash_core.py (hunk 6 above)
Comment (m1raynee) [reply dislash/slash_commands/slash_core.py:487]: You don't pushed it

File: dislash/interactions/message_components.py [hunk 12]
```diff
@@ -1,473 +1,472 @@
...
+                custom_id=custom_id,
+                placeholder=placeholder,
+                min_values=min_values,
+                max_values=max_values,
+                options=options
+            )
+        )
+
+
+MenuOption = SelectOption
```
Comment (m1raynee) [inline dislash/interactions/message_components.py]: What the heck is going on here?

**Review by m1raynee — COMMENTED** [review]
Something strange starts happening here

File: dislash/interactions/message_components.py (hunk 12 above)
Comment (MisileLab) [reply dislash/interactions/message_components.py]: hmm ide broken lol

**Comment by MisileLab** [comment]
I pushed.

File: dislash/interactions/message_components.py (hunk 1 above)
Comment (m1raynee) [reply dislash/interactions/message_components.py:112]: Why did you pass it by? You can do this, here is an example from [discord.py](https://github.com/Rapptz/discord.py/blob/7d9074db8a434995d9e0ae27adb2432b51219c99/discord/enums.py#L310-L349).

File: dislash/slash_commands/slash_core.py [hunk 13]
```diff
@@ -38,9 +37,10 @@
 )
 
 
-#-----------------------------------+
-#              Utils                |
-#-----------------------------------+
+# +-----------------------------------+
+# |              Utils                |
+# +-----------------------------------+
+
```
Comment (m1raynee) [inline dislash/slash_commands/slash_core.py]:
```suggestion
```
Extra line

File: dislash/slash_commands/slash_core.py [hunk 14]
```diff
@@ -54,9 +54,10 @@ def get_class(func):
             return getattr(mod, class_name(func), None)
 
 
-#-----------------------------------+
-#         Core and checks           |
-#-----------------------------------+
+# +-----------------------------------+
+# |         Core and checks           |
+# +-----------------------------------+
+
```
Comment (m1raynee) [inline dislash/slash_commands/slash_core.py]:
```suggestion
```
Extra line

File: dislash/slash_commands/slash_core.py [hunk 15]
```diff
@@ -665,12 +685,13 @@ def cooldown(rate, per, type=BucketType.default):
...
     def decorator(func):
         if isinstance(func, BaseSlashCommand):
             func._buckets = CooldownMapping(Cooldown(rate, per, type))
         else:
             func.__slash_cooldown__ = CooldownMapping(Cooldown(rate, per, type))
         return func
-    return decorator
 
+    return decorator
```
Comment (m1raynee) [inline dislash/slash_commands/slash_core.py]:
```suggestion
    return decorator
```
Extra line

**Comment by MisileLab** [comment]
pushed

**Pull request closed without merging**